flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -303,7 +303,7 @@ private void checkPatternNameUniqueness(final Pattern pattern) {
 				currentPattern = currentPattern.getPrevious();
 
 				final Time currentWindowTime = currentPattern.getWindowTime();
-				if (currentWindowTime != null && currentWindowTime.toMilliseconds() < windowTime) {
+				if (currentWindowTime != null && (windowTime == 0 || currentWindowTime.toMilliseconds() < windowTime)) {
```

Comment:
This check introduces a problem when the user sets the value explicitly to `0`. E.g. `Pattern<Event>begin("start").followedBy("middle").within(Time.seconds(10)).followedBy("then").within(Time.seconds(0)).followedBy("end");` will return as `windowTime` `10_000` because `0` will be interpreted as the default value.